modules/tests/modelator.rs

```diff
@@ -0,0 +1,49 @@
+use eyre::{eyre, Context, Result};
```

Comment:
Could we give a bit more structure to this directory, e.g., put modelator.rs and step.rs inside a module of their own (which we can call `model_based`).

Reply:
Once `modelator.rs` goes way, this directory would only contain `step.rs`. Do you mind if we delay this and try to improve structure in future PRs?